src/Chirp.Web/Pages/TimelineModel.cshtml.cs

```
@@ -9,32 +9,132 @@ public class TimelineModel : PageModel
+        return _likeRepository.LikeExists(authors.First().AuthorId, cheepId);
+    }
+
+    public int GetLikeCount(int cheepId)
+    {
+        return _likeRepository.FindLikeCountByCheepId(cheepId);
+    }
+
+    public bool LikesOwnCheep(string authorName, int cheepId) {
+        var authors = _authorRepository.FindAuthorsByName(authorName);
+        if(!authors.Any()) return false;
+        return _likeRepository.LikesOwnCheep(authors.First().AuthorId, cheepId);
+    }
+
+    public IActionResult OnPostLike(int cheepId)
+    {
+        if (!User.Identity.IsAuthenticated) return Page();
+
+        // Ugly code still needed
+        _authorRepository.CreateAuthor(User.Identity.Name, "example@mail.com");
+
+        int authorId = _authorRepository.FindAuthorsByName(User.Identity.Name).First().AuthorId;
+        _likeRepository.LikeCheep(authorId, cheepId);
+        return RedirectToPage();
+    }
+
+    public IActionResult OnPostUnlike(int cheepId)
+    {
+        if (!User.Identity.IsAuthenticated) return Page();
+
+        // Ugly code still needed
+        _authorRepository.CreateAuthor(User.Identity.Name, "example@mail.com");
+
+        int authorId = _authorRepository.FindAuthorsByName(User.Identity.Name).First().AuthorId;
+        _likeRepository.UnlikeCheep(authorId, cheepId);
+        return RedirectToPage();
+    }
+
+    public bool AuthorFollowsAuthor(string followerName, string followedName)
+    {
+        var followers = _authorRepository.FindAuthorsByName(followerName);
+        if (!followers.Any()) return false; // Should never happen
+
+        var followed = _authorRepository.FindAuthorsByName(followedName);
+        if (!followed.Any()) return false; // Should never happen
+
+        return _followRepository.FollowExists(followers.First().AuthorId, followed.First().AuthorId);
+    }
+
+    public int GetFollowersCount(string routeName)
+    {
+        var authors = _authorRepository.FindAuthorsByName(routeName);
```

You do this check quite frequently. Either put it into a function, or do the check in the author repository. I do however think that we should throw an error if we try to find an author that does not exist.